lib/jsdoc/tag/type.js

```diff
@@ -234,18 +234,24 @@ function getTypeStrings(parsedType, isOutermostType) {
  */
 function parseTypeExpression(tagInfo) {
     let parsedType;
+    let typeExpression = tagInfo.typeExpression;
 
     // don't try to parse empty type expressions
-    if (!tagInfo.typeExpression) {
+    if (!typeExpression) {
         return tagInfo;
     }
 
+    // if expression has typescript import types. Ex: @typedef { import("./").Foo }
+    if (typeExpression.match(/import\(.*\)\./)) {
+        typeExpression = typeExpression.replace(/import\(.*\)\./, '');
```

Comment:
This should be resolving the type not just replacing the string.
For example this doesn't work even though the index file is exporting a default.

```js
/* @typedef { import('./index') } */
```

Reply:
I've managed to bodge the problem by adding a regex check
```js
// if expression has typescript inport types which exported by default. Ex @typedef {import("./foo")}
if (typeExpression.match(/import\(.*\)/)) {
        regex = /import\(['"]?(?:.+)\/([^/\.]+)?(?:\..+)?['"]\)/.exec(typeExpression)
        typeExpression = regex[1]
    }
```
That regex will capture filename of the import. This is very hacky and wont fix the problem if the filename is not the same as the export.    
 Should I do a pull request?

Reply:
I think this would really need to parse the file for it to be usable as the default export may not be named exactly the same.

Reply:
@debugmaster sorry for my delay, I enabled issues only now.

Reply:
Consider improving how `typeExpression` is patched as is done here: https://github.com/polyforest/jsdoc-tsimport-plugin/blob/main/index.js#L134